### PR TITLE
New record and file

### PR DESCRIPTION
**Pull request by Rnhatch**
CStern showed off a cool implementation of content document - where you don't have to have a saved file before using a file attachment.  You upload it to the user record and then reparent it on save.  Cool. 

Elena asked about this on the community today and so I thought I'd expose the cool stuff for the world. A nice side project. 

Can you take a look at the PR?  Thanks.

### CODE REVIEW

File: Salesforce_Advanced/Create_Record_and_File/CreateRecordAndFile.xml [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+			<background type="color" color="white"/>
+		</skuid__wrapper>
+	</components>
+</division>
+</divisions>
+</skuid__grid>
+	</components>
+	<resources>
+		<javascript>
+			<jsitem location="inline" name="newInlineJS" cachelocation="false" url="">(function(skuid){
```
Comment (itmightbecody) [inline Salesforce_Advanced/Create_Record_and_File/CreateRecordAndFile.xml:261]:
```suggestion
			<jsitem location="inline" name="unsavedRecordCheck" cachelocation="false" url="">(function(skuid){
```

Comment (itmightbecody) [reply Salesforce_Advanced/Create_Record_and_File/CreateRecordAndFile.xml:261]: This snippet's purpose is explained in the readme but may as well give it an informative name.

File: Salesforce_Advanced/Create_Record_and_File/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+The process is as follows. 
+- File is initially uploaded as a content document linked to the `current user` record. (Everyone has a saved user record.) 
+- After the new record metadata is completed the save button does the following: 
+    - Saves the new record. 
+    - Creates a new `content link` record connecting the file to the new record that has now been saved. 
+    - Saves the content link record. 
+
+Curiously there is always a content document link record created to the user record. You can't delete it. 
+
+We also include some Javscript and an Action Sequence that deletes the uploaded file if the page is abandoned before the new record is saved.  This keeps things clean.
```
Comment (itmightbecody) [inline Salesforce_Advanced/Create_Record_and_File/README.md:16]:
```suggestion
We also include some JavaScript and an action sequence that deletes the uploaded file if the page is abandoned before the new record is saved.  This keeps things clean. 
```

File: Salesforce_Advanced/Create_Record_and_File/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Create New Record and File in One Step. 
+
+Salesforce content documents and attachments have to be connected to existing records.  This means that you cannot create a new record, and upload a file for that record in the same UI step. Typically Skuid builders have used a 2 step wizard to handle this,  enabling the data entry for a new record, then forcing the user to save that record before they can upload a file or image to it. This is functional,  but not optimal.
```
Comment (itmightbecody) [inline Salesforce_Advanced/Create_Record_and_File/README.md:3]:
```suggestion
Salesforce content documents and attachments have to be connected to **existing records.** This means that you cannot create a new record and upload a file for that record in the same UI step. 

Typically Skuid builders have used a 2 step wizard to handle this, with step one being data entry for a new record, then forcing the user to save that record before they can upload a file or image to it in step two. This is functional, but not optimal. 
```

Comment (itmightbecody) [reply Salesforce_Advanced/Create_Record_and_File/README.md:3]: Not essential, but I think this makes it a little more clear.

File: Salesforce_Advanced/Create_Record_and_File/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Create New Record and File in One Step. 
+
+Salesforce content documents and attachments have to be connected to existing records.  This means that you cannot create a new record, and upload a file for that record in the same UI step. Typically Skuid builders have used a 2 step wizard to handle this,  enabling the data entry for a new record, then forcing the user to save that record before they can upload a file or image to it. This is functional,  but not optimal. 
+
+This sample page shows how to create a case, and upload a file to that case in the same step. Though this method seems like a bit of hack - its actually used behind the scenes by Salesforce in some of their standard UI.
```
Comment (itmightbecody) [inline Salesforce_Advanced/Create_Record_and_File/README.md:5]:
```suggestion
This sample page shows how to create a case and upload a file to that case in the same step. Though this method seems like a bit of hack, it's actually used behind the scenes by Salesforce in some of their standard UI. 
```
Some grammar things.

File: Salesforce_Advanced/Create_Record_and_File/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Create New Record and File in One Step. 
+
+Salesforce content documents and attachments have to be connected to existing records.  This means that you cannot create a new record, and upload a file for that record in the same UI step. Typically Skuid builders have used a 2 step wizard to handle this,  enabling the data entry for a new record, then forcing the user to save that record before they can upload a file or image to it. This is functional,  but not optimal. 
+
+This sample page shows how to create a case, and upload a file to that case in the same step. Though this method seems like a bit of hack - its actually used behind the scenes by Salesforce in some of their standard UI. 
+
+The process is as follows.
```
Comment (itmightbecody) [inline Salesforce_Advanced/Create_Record_and_File/README.md:7]:
```suggestion
The process is as follows:
```

**Comment by Rnhatch** [comment]
Thanks Cody!

**Pull request merged**